File: HealthLink_MB/lib/models/pharmacy/pharmacy_order_item.dart
class PharmacyOrderItem {
  final int? orderItemId;
  final int? medicineId;
  final String medicationName;
  final int? totalSupplyDays;
  final int quantity;
  final String? unit;
  final String? frequency;
  final String? timing;
  final String? route;
  final double? unitPrice;
  final double? totalPrice;
  final String? notes;

  const PharmacyOrderItem({
    this.orderItemId,
    this.medicineId,
    required this.medicationName,
    this.totalSupplyDays,
    required this.quantity,
    this.unit,
    this.frequency,
    this.timing,
    this.route,
    this.unitPrice,
    this.totalPrice,
    this.notes,
  });

  factory PharmacyOrderItem.fromJson(Map<String, dynamic> json) {
    return PharmacyOrderItem(
      orderItemId: json['orderItemId'] as int?,
      medicineId: json['medicineId'] as int?,
      medicationName: json['medicationName'] as String? ?? '',
      totalSupplyDays: json['totalSupplyDays'] as int?,
      quantity: json['quantity'] as int? ?? 1,
      unit: json['unit'] as String?,
      frequency: json['frequency'] as String?,
      timing: json['timing'] as String?,
      route: json['route'] as String?,
      unitPrice: (json['unitPrice'] as num?)?.toDouble(),
      totalPrice: (json['totalPrice'] as num?)?.toDouble(),
      notes: json['notes'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'medicineId': medicineId,
      'medicationName': medicationName,
      'totalSupplyDays': totalSupplyDays,
      'quantity': quantity,
      'unit': unit,
      'frequency': frequency,
      'timing': timing,
      'route': route,
      'unitPrice': unitPrice,
    };
  }
}

File: HealthLink_BE/src/main/java/com/HealthLink/entity/PharmacyOrderItem.java
package com.HealthLink.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.math.BigDecimal;

@Entity
@Table(name = "PharmacyOrderItems")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PharmacyOrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "OrderItemID")
    private Integer orderItemId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "OrderID", nullable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private PharmacyOrder pharmacyOrder;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "MedicineID")
    @ToString.Exclude
    private Medicine medicine;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "SourcePrescriptionHeaderID")
    @ToString.Exclude
    private PrescriptionHeader sourcePrescriptionHeader;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "SourcePrescriptionItemID")
    @ToString.Exclude
    private PrescriptionItem sourcePrescriptionItem;

    @Column(nullable = false, length = 200)
    private String medicationName;

    @Column(nullable = false)
    private Integer totalSupplyDays;

    @Column(nullable = false)
    private Integer quantity;

    @Column(length = 50)
    private String unit;

    @Column(length = 100)
    private String frequency;

    @Column(length = 100)
    private String timing;

    @Column(length = 50)
    private String route;

    private BigDecimal totalPrice;

    @Column(length = 500)
    private String notes;
}
